nav2_util/src/robot_utils.cpp

```diff
@@ -13,107 +14,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <nav2_util/robot_utils.hpp>
+#include "nav2_util/robot_utils.hpp"
+#include "rclcpp/logger.hpp"
 
 namespace nav2_util
 {
 
-VelocityPublisher::VelocityPublisher(
-  rclcpp::Node::SharedPtr & node,
-  const std::string topic)
-{
-  vel_pub_ = node->create_publisher<geometry_msgs::msg::Twist>(topic, 1);
-}
-
-void
-VelocityPublisher::publishCommand(const geometry_msgs::msg::Twist & cmd_vel)
-{
-  vel_pub_->publish(cmd_vel);
-}
-
-RobotStateHelper::RobotStateHelper(
-  rclcpp::Node::SharedPtr & node,
-  const std::string odom_topic)
-: node_(node)
-{
-  pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
-    "amcl_pose", std::bind(&RobotStateHelper::onPoseReceived, this, std::placeholders::_1));
-
-  odom_sub_ = node_->create_subscription<nav_msgs::msg::Odometry>(
-    odom_topic, std::bind(&RobotStateHelper::onOdomReceived, this, std::placeholders::_1));
-
-  initial_odom_received_ = false;
-  initial_pose_received_ = false;
-}
-
-bool
-RobotStateHelper::getOdometry(nav_msgs::msg::Odometry::SharedPtr & robot_odom)
-{
-  std::shared_lock<std::shared_timed_mutex> lock(state_mutex_);
-  if (!initial_odom_received_) {
-    RCLCPP_DEBUG(node_->get_logger(),
-      "getOdometry: Can't return current velocity: Initial odometry not yet received.");
-    return false;
-  }
-
-  robot_odom = current_odom_;
-  return true;
-}
-
-bool
-RobotStateHelper::getCurrentPose(
-  geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr & robot_pose,
-  const bool use_topic)
-{
-  if (use_topic) {
-    return getGlobalLocalizerPose(robot_pose);
+bool getCurrentPose(
+  geometry_msgs::msg::PoseStamped & global_pose,
+  std::shared_ptr<tf2_ros::Buffer> & tf_buffer, const std::string global_frame,
+  const std::string robot_frame, const double transform_timeout)
+{
+  static rclcpp::Logger logger = rclcpp::get_logger("getCurrentPose");
```

Comment:
just curious why is `static`?

Reply:
Performance, otherwise, every time its called a new `rclcpp::Logger` object would be made and that's going to have a bit of overhead for a function that could very realistically be called at 100hz. This way we incur some cost the first time its called in a process and afterwards logging is "free" without having to pass a pointer of the caller node which seemed ugly. 

... also maybe a reason for a class object but this meets my "i'm ok with it" by a razors edge for leaving it as a function. 

It wouldn't need to be static as an object. Maybe I should think about this one again.